packages/cli/src/oclif/commands/scaffold.js

```diff
@@ -0,0 +1,217 @@
+const _ = require('lodash');
+const path = require('path');
+const colors = require('colors/safe');
+
+const { flags } = require('@oclif/command');
+
+const BaseCommand = require('../ZapierBaseCommand');
+const { buildFlags } = require('../buildFlags');
+const utils = require('../../utils');
+
+class ScaffoldCommand extends BaseCommand {
+  writeTemplateFile(templatePath, templateContext, dest) {
+    const destPath = path.join(process.cwd(), `${dest}.js`);
+    const preventOverwrite = !this.flags.force;
+
+    if (preventOverwrite && utils.fileExistsSync(destPath)) {
+      const [location, filename] = dest.concat('.js').split('/');
+      return Promise.reject(
+        [
+          `File ${colors.bold(filename)} already exists within ${colors.bold(
+            location
+          )}.`,
+          'You could:',
+          '1. Choose a different filename',
+          `2. Delete ${filename} from ${location}`,
+          `3. Run ${colors.italic('scaffold')} with ${colors.bold(
+            '--force'
+          )} to overwrite the current ${filename}`
+        ].join('\n')
+      );
+    }
+
+    return utils
+      .readFile(templatePath)
+      .then(templateBuf => templateBuf.toString())
+      .then(template =>
+        _.template(template, { interpolate: /<%=([\s\S]+?)%>/g })(
+          templateContext
+        )
+      )
+      .then(rendered => {
+        this.startSpinner(`Writing new ${dest}.js`);
+        return utils
+          .ensureDir(path.dirname(destPath))
+          .then(() => utils.writeFile(destPath, rendered));
+      })
+      .then(this.stopSpinner);
+  }
+
+  async perform() {
+    const { name, type } = this.args;
+    const templateContext = {
```

Comment:
organization wise, I might move all of this boilerplate into a function that takes `name` to keep the command itself a little cleaner.

Reply:
That makes sense. I wasn't sure how much of the command to re-write in porting you were going for. This helps!